### PR TITLE
Reading list messages updated to include the list name

### PR DESCRIPTION
**Pull request by sharvaniharan**
**Phab:** https://phabricator.wikimedia.org/T275604

### CODE REVIEW

File: app/src/main/java/org/wikipedia/readinglist/ReadingListBehaviorsUtil.kt [hunk 1]
```diff
@@ -149,7 +149,7 @@ object ReadingListBehaviorsUtil {
         FeedbackUtil
                 .makeSnackbar(activity,
                         activity.getString(
-                                if (lists.size == 1) R.string.reading_list_item_deleted else R.string.reading_lists_item_deleted, page.title()),
+                                if (lists.size == 1) R.string.reading_list_item_deleted_from_list else R.string.reading_lists_item_deleted, page.title()),
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/readinglist/ReadingListBehaviorsUtil.kt:152]: The first one will cause a crash since it has two parameters.

**Review by cooltey — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/wikipedia/readinglist/ReadingListBehaviorsUtil.kt (hunk 1 above)
Comment (sharvaniharan) [reply app/src/main/java/org/wikipedia/readinglist/ReadingListBehaviorsUtil.kt]: Thank you for catching this @cooltey. Done.

File: app/src/main/java/org/wikipedia/readinglist/ReadingListBehaviorsUtil.kt [hunk 2]
```diff
@@ -146,14 +148,17 @@ object ReadingListBehaviorsUtil {
...
         }
-        FeedbackUtil
-                .makeSnackbar(activity,
-                        activity.getString(
-                                if (lists.size == 1) R.string.reading_list_item_deleted else R.string.reading_lists_item_deleted, page.title()),
-                        FeedbackUtil.LENGTH_DEFAULT)
+        val readingListNames = mutableListOf<String>()
+        lists.forEach { readingList ->
+            readingListNames.add(readingList.title())
+        }
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/readinglist/ReadingListBehaviorsUtil.kt:154]: Would it be possible to simplify as below and not running the logic in the snackbar?
```
        val readingListNames = lists.map { it.title() }.run {
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
                ListFormatter.getInstance().format(this)
            } else {
                joinToString(separator = ", ")
            }
        }
```

File: app/src/main/java/org/wikipedia/readinglist/LongPressMenu.kt [hunk 3]
```diff
@@ -85,10 +87,16 @@ class LongPressMenu(private val anchorView: View, private val existsInAnyList: B
...
         listsContainingPage?.let { list ->
-            RemoveFromReadingListsDialog(list).deleteOrShowDialog(anchorView.context) { _: List<ReadingList?>?, _: ReadingListPage? ->
+            RemoveFromReadingListsDialog(list).deleteOrShowDialog(anchorView.context) { readingLists, _ ->
                 entry?.let {
                     if (anchorView.isAttachedToWindow) {
-                        showMessage((anchorView.context as AppCompatActivity), anchorView.context.getString(R.string.reading_list_item_deleted, it.title.displayText))
+                        val readingListNames = mutableListOf<String>()
+                        readingLists.forEach { readingList ->
+                            readingListNames.add(readingList.title())
+                        }
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/readinglist/LongPressMenu.kt:96]: This could be simplified (see comment below).

**Review by cooltey — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/wikipedia/readinglist/LongPressMenu.kt (hunk 3 above)
Comment (cooltey) [reply app/src/main/java/org/wikipedia/readinglist/LongPressMenu.kt:96]: Is this resolved?

Comment (sharvaniharan) [reply app/src/main/java/org/wikipedia/readinglist/LongPressMenu.kt:96]: yes @cooltey

Comment (sharvaniharan) [reply app/src/main/java/org/wikipedia/readinglist/LongPressMenu.kt]: Sorry resolved now

**Pull request merged**